runtime/guard-jwt/src/main/java/io/aklivity/zilla/runtime/guard/jwt/internal/JwtGuardHandler.java

```diff
@@ -54,39 +63,43 @@ public class JwtGuardHandler implements GuardHandler
     private final Long2ObjectHashMap<JwtSessionStore> sessionStoresByContextId;
 
     public JwtGuardHandler(
-        JwtOptionsConfig options,
-        LongSupplier supplyAuthorizedId)
+            GuardConfig guard,
+            LongSupplier supplyAuthorizedId)
     {
+
+        JwtOptionsConfig options = (JwtOptionsConfig) guard.options;
         this.issuer = options.issuer;
         this.audience = options.audience;
         this.challenge = options.challenge.orElse(null);
 
-        Map<String, JsonWebKey> keys = new HashMap<>();
+        Map<String, JsonWebKey> tmpKeys = new HashMap<>();
         if (options.keys != null)
         {
-            for (JwtKeyConfig key : options.keys)
+            tmpKeys = getKeys(options.keys);
+        }
+        else
+        {
+            String keysURL = options.keysURL.orElse(getJwksURL(issuer));
+            if (keysURL != null)
             {
                 try
                 {
-                    Map<String, Object> params = new HashMap<>();
-                    params.put("kty", key.kty);
-                    params.put("kid", key.kid);
-                    params.put("e", key.e);
-                    params.put("n", key.n);
-                    params.put("alg", key.alg);
-                    params.put("crv", key.crv);
-                    params.put("x", key.x);
-                    params.put("y", key.y);
-                    params.put("use", key.use);
-                    keys.put(key.kid, JsonWebKey.Factory.newJwk(params));
+                    String keysText = guard.readURL.apply(URI.create(keysURL).toURL());
+                    JsonArray keysArray = guard.jsonb.fromJson(keysText, JsonArray.class);
+                    List<JwtKeyConfig> keysConfig = keysArray
+                            .stream()
+                            .map(JsonValue::asJsonObject)
+                            .map(new JwtKeyConfigAdapter()::adaptFromJson)
+                            .collect(toList());
+                    tmpKeys = getKeys(keysConfig);
                 }
-                catch (JoseException ex)
+                catch (MalformedURLException | JsonbException ignored)
```

Comment:
Better to let this exception escape so we can tell what's wrong instead of hiding it.